app_test.go

```diff
@@ -262,149 +261,270 @@ func TestBashCompletionOptions(t *testing.T) {
 		return []string{"arg-4-opt-1", "arg-4-opt-2"}
 	}).String()
 
+	four := a.Command("four", "")
+	four.Arg("arg-5", "").ExistingFileOrDir()
+	four.Arg("arg-6", "").HintOptions("/usr", "/usr/local").ExistingDir()
+
+	five := a.Command("five", "")
+	five.Flag("flag-5", "").ExistingFile()
+	five.Command("sub-1", "").Flag("flag-6", "").ExistingDir()
+
 	cases := []struct {
-		Args            string
-		ExpectedOptions []string
+		Args          string
+		ExpectedFlags []string
+		ExpectedWords []string
 	}{
 		{
-			Args:            "--completion-bash",
-			ExpectedOptions: []string{"help", "one", "three", "two"},
+			Args:          "--completion-bash",
+			ExpectedFlags: []string(nil),
+			ExpectedWords: []string{"five", "four", "help", "one", "three", "two"},
 		},
 		{
-			Args:            "--completion-bash --",
-			ExpectedOptions: []string{"--flag-0", "--flag-1", "--help"},
+			Args:          "--completion-bash --",
+			ExpectedFlags: []string(nil),
+			ExpectedWords: []string{"--flag-0", "--flag-1", "--help"},
```

Comment:
So a rough translation of the current nomenclature would be
* `flags`: flags that ought to be passed directly to `compgen` (e.g. `-d`, `-f`)
* `words`: words that ought to be added to the wordlist of `compgen`'s `-W` flag

Since you're (reasonable) asking for this to be public, I'll add some documentation.  If you think there is nomenclature that is clearer, I'm of course happy to make a change.